app/src/main/java/com/appunite/loudius/ui/login/LoginScreen.kt

```diff
@@ -60,6 +95,28 @@ fun LoginScreen() {
             },
 
         )
+        if (state.showXiaomiPermissionDialog) {
+            LoudiusDialog(
+                onDismissRequest = { onAction(LoginAction.XiaomiPermissionDialogDismiss) },
+                title = stringResource(R.string.login_screen_xiaomi_dialog_title),
+                text = {
+                    LoudiusText(
+                        style = LoudiusTextStyle.ScreenContent,
+                        text = stringResource(R.string.login_screen_xiaomi_dialog_text),
+                    )
+                },
+                confirmButton = {
+                    LoudiusOutlinedButton(text = stringResource(R.string.login_screen_xiaomi_dialog_grant_permission)) {
+                        onAction(LoginAction.XiaomiPermissionDialogGrantPermission)
+                    }
+                },
+                dismissButton = {
+                    LoudiusOutlinedButton(text = stringResource(R.string.login_screen_xiaomi_dialog_already_granted)) {
+                        onAction(LoginAction.XiaomiPermissionDialogAlreadyGrantedPermission)
+                    }
+                },
+            )
+        }
```

Comment:
**Suggestion:** Extract that into separate composable for example `XiaomiPermissionDialog` that would be faster to read at first glance.